### PR TITLE
CASEFLOW-808 Add more actions during CAVC-Remand-Processed-Letter Response Window task

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves [CASEFLOW-808](https://vajira.max.gov/browse/CASEFLOW-808)

As a CAVC Litigation Support user, I want the ability to perform actions while the CAVC-Remand-Processed-Letter Response Window Task is on hold, so that I can act on the response from the Veteran before the 90 day window is complete.
If I end the hold, I can put the task back on hold for the completion of that window.

### Description
Add more actions to the CAVC-Remand-Processed-Letter Response Window task and adjust when the actions are available for different scenarios.

### Acceptance Criteria
- [x] Code compiles correctly
- [x] Given the user is apart of the CAVC Litigation Support team, when they view the Case Details page of their assigned CAVC-Remand-Processed-Letter Response Window Task and click the Actions dropdown, then the end hold early as well as all available actions when ending the hold early is available
- [x] Verify the On-Hold Window task continues to display in the Org Assigned Tab
- [x] User cannot mark task complete without ending the hold
- [x] User would not have to end window task hold and do not have to recalculate # of days left. Ensure current org tabs remain the same.

Updated AC, explained in this [thread](https://dsva.slack.com/archives/CJL810329/p1612966041232900):
- [x] "Mark task complete" and "Extension request" should not be available unless the window task is first assigned to a user

### Testing Plan
Test the following for both CAVC admin and non-admin users.

1. Login as a CAVC lit support user
2. Go to http://localhost:3000/organizations/cavc-lit-support?tab=unassignedTab
3. Select an appeal with "Send CAVC-Remand-Processed Letter Task" to view the Case Details page. Note the docket number and UUID so you can find this appeal later.
4. Select "Mark as complete" and submit
5. Go to the team/org view; click the "Assigned" tab to find the appeal; click the appeal
6. Note the 5 actions to assign to other teams are available while the task is on hold
7. Select and submit "End hold early"
8. Note the 5 actions to assign to other teams are STILL available while the task is NOT on hold.
9. Note "Mark task complete" is now available, i.e., when the task is NOT on hold. (We'll test this near the end.)
10. Play around assigning tasks to other teams with and without the window task being on hold. Check the task tree and the appeals location in the correct queue tab. Verify the On-Hold Window task continues to display in the Org Assigned Tab
11. Try "Assign to person" and check the task tree. Note that the timed hold is not migrated/copied to the user task (bug ticket #15893) -- workaround: manually put the task on hold. Play around assigning tasks to other teams with and without the window task being on hold. Check the task tree and the appeals location in the correct queue tab. Verify the On-Hold Window task continues to display in the Org Assigned Tab. See screenshots below for the expected task actions when a task is assigned to person.
12. End the hold and click "Mark task complete" and submit.

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
(on hold; time window is open) image | (when not assigned to user) image<br/> (when assigned to user) image
(not on hold; time window ended) image | (when not assigned to user) image <br/> (when assigned to user) image

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: spec/factories/cavc_remand.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
         JudgeTeam.first&.non_admins&.first ||
           create(:user).tap { |u| create(:staff, :attorney_role, user: u) }
       end
-      veteran { Veteran.first || create(:veteran) }
+      veteran { create(:veteran) }
```
Comment (yoomlam) [inline spec/factories/cavc_remand.rb:22]: Results in each CAVC appeal have a different Veteran, so that all CAVC appeals are for the same veteran.

**Comment by araposo-tistatech** [comment]
PM Reviewed!

File: app/models/tasks/send_cavc_remand_processed_letter_task.rb [hunk 2]
```diff
@@ -55,7 +55,7 @@ def available_actions(user)
 
   def update_from_params(params, current_user)
     if params[:status] == "completed"
-      # Create ResponseWindowTask before completing this task so that parent CavcTask is remains on-hold
+      # Create ResponseWindowTask before completing this task so that parent CavcTask remains on-hold
```
Comment (n1zyy) [inline app/models/tasks/send_cavc_remand_processed_letter_task.rb:58]: 😍 for taking the time to make these little things better as you go.

**Review by n1zyy — COMMENTED** [review]
Code looks good! About to run through testing now.

**Review by ajspotts — APPROVED** [review]
looks good lgtm!

**Pull request merged**